code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/BasicNotificationBuilderTest.kt

```diff
@@ -0,0 +1,141 @@
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import kotlin.test.assertNotNull
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import junit.framework.TestCase.assertEquals
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+import android.content.Intent
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import org.junit.Assert.*
+import org.robolectric.Shadows.shadowOf
+import kotlin.properties.Delegates
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class BasicNotificationBuilderTest {
+
+    @Mock
+    private lateinit var basicPushTemplate: BasicPushTemplate
+    private lateinit var context: Context
+    private lateinit var trackerActivityClass: Class<out Activity>
+
+
+    private lateinit var dataMap: MutableMap<String, String>
```

Comment:
dataMap is not modified after creation so this does not need to be a MutableMap

Reply:
Modified code to use data from the newly added utility class.